### ISSUE TITLE
Can't enter custom listing credits amount

### ISSUE THREAD

**Issue by GMartigny** [thepracticaldev/dev.to#2566]
**Describe the bug**
In the buying credits for listing page, the amount input can't be filled using number key on keyboard.
On the JS code, there's:
```js
document.getElementById("amount-input").addEventListener("keydown",function (e) {
	if ((e.keyCode < 48 || e.keyCode > 57) && e.keyCode != 8) {
		e.preventDefault();
	}
	calculatePriceAndShow();
});
```
The `preventDefault` function is wrongfully called. I don't get what it's used for anyway.

**To Reproduce**

1. Go to https://dev.to/credits/new
2. Click on 'Amount to purchase' input
3. Try entering custom amount using keyboard numbers
4. Nothing append

**Expected behavior**
I should be able to enter any amount.

**Comment by mjraadi** [comment]
@GMartigny the condition is there to only allow English numbers. If it's not there you can input any number from other languages. I was able to put ۱۲۳۴۵۶۷۸۹۰(which is 1234567890 in Farsi) without the condition and it's still valid number. Take a look below:
![Screenshot_20190503_144503](https://user-images.githubusercontent.com/20579660/57131822-e24fd600-6db2-11e9-8f61-8c1812c3978b.png)

However, the condition won't allow for keypad numbers because keypad 0-9 is keycode `96` to `105`. I believe the correct function should be this:
```javascript
function listenForNumberChange() {
  document.getElementById("amount-input").addEventListener("keydown",function (e) {
    (
      (e.keyCode >= 48 && e.keyCode <= 57) || // Keyboard 0-9 Keycode is 48 to 57
      (e.keyCode >= 96 && e.keyCode <= 105) || //Keypad 0-9 Keycode is 96 to 105
      (e.keyCode == 8) //Allow backspace
      
    ) ? 
      calculatePriceAndShow():
      e.preventDefault();
  });
}
```

**Comment by GMartigny** [comment]
Ok, but why prevent typing farsi numbers ? JS seems to understand the input value correctly and display the right amount (3086419725 $) on the right of the input.

![image](https://user-images.githubusercontent.com/2543511/57136608-9a7a8000-6dad-11e9-8d03-4ce08888601c.png)

**IMO**, the website should not prevent user actions, but they should be sanitized on the back-end.

**Comment by mariocsee** [comment]
resolved with PR #2672
